File: Claude/1.Algoritmo Básico Ordenamiento por Burbuja/golang/burbuja.go
package main

import "fmt"

// BubbleSort implementa el algoritmo de ordenamiento burbuja
func BubbleSort(arr []int) []int {
    // Crear copia del slice para no modificar el original
    sorted := make([]int, len(arr))
    copy(sorted, arr)
    
    n := len(sorted)
    
    for i := 0; i < n-1; i++ {
        swapped := false
        for j := 0; j < n-i-1; j++ {
            if sorted[j] > sorted[j+1] {
                sorted[j], sorted[j+1] = sorted[j+1], sorted[j]
                swapped = true
            }
        }
        if !swapped {
            break
        }
    }
    
    return sorted
}

func main() {
    numbers := []int{64, 34, 25, 12, 22, 11, 90}
    sorted := BubbleSort(numbers)
    
    fmt.Printf("Array original: %v\n", numbers)
    fmt.Printf("Array ordenado: %v\n", sorted)
}

File: Claude/1.Algoritmo Básico Ordenamiento por Burbuja/python/burbuja.py
def bubble_sort(arr):
    """
    Implementación del algoritmo de ordenamiento burbuja.
    Complejidad temporal: O(n²)
    Complejidad espacial: O(1)
    """
    n = len(arr)
    for i in range(n):
        swapped = False
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped:
            break
    return arr

# Ejemplo de uso
numbers = [64, 34, 25, 12, 22, 11, 90]
sorted_numbers = bubble_sort(numbers.copy())
print(f"Array original: {numbers}")
print(f"Array ordenado: {sorted_numbers}")